React-Template-master/src/components/MessageModal.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState } from 'react';
+import { Modal, Box, Typography, Button, useTheme } from '@mui/material';
+import Correct from '../assets/Correct.svg';
+import Warning from '../assets/Warning.svg';
+import { Logo } from './Logo';
+import { useTranslation } from 'react-i18next';
+
+export type SupportedModalType = 'correct' | 'warning';
+export interface ModalContent {
+    title: string;
+    subtitle: string;
+    type?: SupportedModalType;
+    buttonText: string;
+}
+
+interface MessageModalProps {
+    open: boolean;
+    setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+    modalContent: ModalContent;
+}
+
+const style = {
+    display: 'flex',
+    alignItems: 'center',
+    flexDirection: 'column',
+    border: 'none',
+    position: 'absolute',
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    width: '408px',
+    borderRadius: '4px',
+    height: '268px',
+    boxShadow: 24,
+    p: '20px',
+    boxSizeing: 'content-box'
+};
+
+export const MessageModal = ({
+    open,
+    setOpen,
+    modalContent: { type, title, subtitle, buttonText }
+}: MessageModalProps): JSX.Element => {
+    const { t } = useTranslation();
+    const theme = useTheme();
+    const isCorrect = type === 'correct';
+    return (
+        <Box>
+            <Modal
+                open={open}
+                onClose={(): void => {
+                    setOpen(false);
+                }}>
+                <Box sx={{ ...style, bgcolor: theme.palette.common.white }}>
+                    {type &&
+                        (isCorrect ? (
+                            <Logo type={'modal'} src={Correct} />
+                        ) : (
+                            <Logo type={'modal'} src={Warning} />
+                        ))}
+                    <Typography variant="subtitle1" sx={{ mt: '11.3px' }}>
+                        {t(title)}
```

Comment:
Instead of sending the key it would better to send the translated text to the component.
In that case the parent will decide what the text is shown instead of the child component deciding what the text will be.

Reply:
okay